algonaut_transaction/src/api_model.rs

```diff
@@ -210,6 +229,14 @@ impl TryFrom<ApiTransaction> for Transaction {
 
                     extra_pages: num_from_api_option(api_t.extra_pages),
                 })
+            },
+            "stpf"=>{
+                TransactionType::StateProofTransaction(StateProofTransaction {
+                    sender: api_t.sender,
+                    state_proof_type: StateProofType::StateProofBasic,
+                    //state_proof: api_t.state_proof,
```

Comment:
I'd use here a function similar to [parse_asset_transfer_transaction](https://github.com/manuelmauro/algonaut/blob/5b395102e6dc0d03ba67ba00e19dd4fac7e587ff/algonaut_transaction/src/api_model.rs#L259-L298), that creates the domain object if all the optionals are set and errors otherwise.